flow/energy_models/power_demand.py

```diff
@@ -0,0 +1,57 @@
+import math
+import random
+import statistics
+import numpy as np
+from scipy.interpolate import interp1d
+from collections import namedtuple
+from flow.energy_models.base_energy import BaseEnergyModel
+from abc import ABCMeta, abstractmethod
+
+class PowerDemandModel(BaseEnergyModel, metaclass=ABCMeta):
+    """Calculate power consumption of a vehicle.
+    Assumes vehicle is an average sized vehicle.
+    The power calculated here is the lower bound of the 
+    actual power consumed by the vehicle
+    """
+
+    def __init__(self, kernel, mass=2041, area=3.2, Cr=0.0027, Ca=0.4):
```

Comment:
We probably want these parameters to be lower-case variables for consistent styling purposes. you may as well just call them `rolling_res_coeff` and `aerodynamic_drag_coeff`